artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/AddressInfo.java

```diff
@@ -191,4 +194,16 @@ public long getUnRoutedMessageCount() {
       return unRoutedMessageCountUpdater.get(this);
    }
 
+   public void registerMeters(MetricsManager metricsManager) {
+      if (metricsManager != null) {
+         metricsManager.registerAddressGauge("routed.message.count", name.toString(), this, metrics -> Double.valueOf(getRoutedMessageCount()), AddressControl.ROUTED_MESSAGE_COUNT_DESCRIPTION);
+         metricsManager.registerAddressGauge("unrouted.message.count", name.toString(), this, metrics -> Double.valueOf(getUnRoutedMessageCount()), AddressControl.ROUTED_MESSAGE_COUNT_DESCRIPTION);
+      }
+   }
+
+   public void unregisterMeters(MetricsManager metricsManager) {
+      if (metricsManager != null) {
```

Comment:
Given that `metricsManager` is volatile, it should be called in a local variable and if not null, used to call its method

Reply:
Ups sorry, reading it from the web I haven't noticed from where `metricsManager` was coming